hnn_core/network.py

```diff
@@ -315,44 +314,57 @@ def _create_all_src(self):
             if pc.gid_exists(gid):
                 # get type of cell and pos via gid
                 # now should be valid for ext inputs
-                type = self.gid_to_type(gid)
-                type_pos_ind = gid - self.gid_dict[type][0]
-                pos = self.pos_dict[type][type_pos_ind]
+                cell_type = self.gid_to_type(gid)
+                type_pos_ind = gid - self.gid_dict[cell_type][0]
+                pos = self.pos_dict[cell_type][type_pos_ind]
                 # figure out which cell type is assoc with the gid
                 # create cells based on loc property
                 # creates a NetCon object internally to Neuron
                 type2class = {'L2_pyramidal': L2Pyr, 'L5_pyramidal': L5Pyr,
                               'L2_basket': L2Basket, 'L5_basket': L5Basket}
-                if type in ('L2_pyramidal', 'L5_pyramidal', 'L2_basket',
-                            'L5_basket'):
-                    Cell = type2class[type]
-                    if type in ('L2_pyramidal', 'L5_pyramidal'):
+                # first check for actual cell types
+                if cell_type in ('L2_pyramidal', 'L5_pyramidal', 'L2_basket',
+                                 'L5_basket'):
+                    Cell = type2class[cell_type]
+                    if cell_type in ('L2_pyramidal', 'L5_pyramidal'):
                         self.cells.append(Cell(gid, pos, self.params))
                     else:
                         self.cells.append(Cell(gid, pos))
                     pc.cell(
                         gid, self.cells[-1].connect_to_target(
                             None, self.params['threshold']))
-                elif type == 'extinput':
-                    # print('type',type)
+                # external inputs are special types of pseudo-cells
+                elif cell_type == 'common':
+                    # print('cell_type',cell_type)
                     # to find param index, take difference between REAL gid
                     # here and gid start point of the items
-                    p_ind = gid - self.gid_dict['extinput'][0]
+                    p_ind = gid - self.gid_dict['common'][0]
+
+                    # new ExtFeed: cell type irrelevant (None) since input
+                    # timing is unaffected by it; cell_type == feed_type
+                    self.common_feed_list.append(
```

Comment:
```suggestion
                    self.common_feeds.append(
```
It's a bit superfluous to say common_feed_list when you see `append`

Reply:
Sure, but reason was to distinguish it from `unique_feed_dict`, which is a `dict`! It's quite confusing that there are two containers for what seems to be similar things. I thought it helped around L345 and L363 to keep tabs on what's being sent to `ExtFeed`. I'll take another look though

Reply:
It should be OK